src/falcon/falcon_c/rpo.h

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <string.h>
+
+/* ==================================================================== */
+/*
+ * RPO hashing algorithm related structs and methods.
+ */
+
+typedef struct
+{
+    union
+    {
+        uint64_t A[12];
+        uint8_t dbuf[96];
+    } st;
+    uint64_t dptr;
+} inner_rpo128_context;
+
+void rpo128_inc_init(inner_rpo128_context *rc);
+void rpo128_inc_absorb(inner_rpo128_context *rc, const uint8_t *in, size_t len);
+void rpo128_inc_finalize(inner_rpo128_context *rc);
+void rpo128_inc_squeeze(inner_rpo128_context *rc, uint8_t *out, size_t len);
+
+#define inner_rpo128_init(rc) rpo128_inc_init(rc)
+#define inner_rpo128_inject(rc, in, len) rpo128_inc_absorb(rc, in, len)
+#define inner_rpo128_flip(rc) rpo128_inc_finalize(rc)
+#define inner_rpo128_extract(rc, out, len) rpo128_inc_squeeze(rc, out, len)
+
+/* ==================================================================== */
+/*
+ * Hash-to-Point algorithm for signature generation and signature
+ * verification.
+ */
+
+void PQCLEAN_FALCON512_CLEAN_hash_to_point_rpo(inner_rpo128_context *rc,
+                                                       uint16_t *x, unsigned logn);
```

Comment:
Would be great to add brief comments explaining what this function does (e.g., "it generates n random values based on the input seed x` or something like that).